common/src/main/java/tachyon/security/LoginUser.java

```diff
@@ -0,0 +1,122 @@
+
+/**
+ * A Singleton of LoginUser, which is an instance of {@link tachyon.security.User}. It represents
+ * the user of Tachyon client, when connecting to Tachyon service.
+ *
+ * The implementation of getting a login user supports Windows, Unix, and Kerberos login modules.
+ *
+ * This singleton uses lazy initialization.
+ */
+public class LoginUser {
+
+  /** User instance of the login user in Tachyon client process */
+  private static User sLoginUser;
+
+  /**
+   * This constructor should never be invoked.
+   */
+  private LoginUser() {
+
+  }
+
+  /**
+   * Get current singleton login user.
+   * This method is called to identify the singleton user who runs Tachyon client.
+   * When Tachyon client gets a user by this method and connects to Tachyon service,
+   * this user represents the client and is maintained in service.
+   * @return the login user
+   * @throws java.io.IOException if login fails
+   */
+  public static User get() throws IOException {
+    return get(new TachyonConf());
+  }
+
+  /**
+   * Get current singleton login user.
+   * This method is called to identify the singleton user who runs Tachyon client.
+   * When Tachyon client gets a user by this method and connects to Tachyon service,
+   * this user represents the client and is maintained in service.
+   * @param conf Tachyon configuration
+   * @return the login user
+   * @throws java.io.IOException if login fails
+   */
+  private static User get(TachyonConf conf) throws IOException {
+    if (sLoginUser == null) {
+      synchronized (LoginUser.class) {
+        if (sLoginUser == null) {
+          sLoginUser = login(conf);
+        }
+      }
+    }
+    return sLoginUser;
+  }
+
+  /**
+   * Login based on the LoginModules
+   * @param conf Tachyon configuration
+   * @return the login user
+   * @throws IOException if login fails
+   */
+  private static User login(TachyonConf conf) throws IOException {
+    AuthenticationFactory.AuthType sAuthType = AuthenticationFactory.getAuthTypeFromConf(conf);
+    isSecurityEnabled(sAuthType);
+
+    try {
+      Subject subject = new Subject();
+
+      LoginContext loginContext = new LoginContext(sAuthType.getAuthName(), subject, null,
+          new TachyonJaasConfiguration());
+      loginContext.login();
+
+      Set<User> userSet = subject.getPrincipals(User.class);
+      if (!userSet.isEmpty()) {
+        if (userSet.size() == 1) {
+          return userSet.iterator().next();
+        } else {
+          throw new LoginException("More than one Tachyon User is found");
+        }
+      } else {
+        throw new LoginException("No Tachyon User is found.");
+      }
+    } catch (LoginException e) {
+      throw new IOException("Fail to login", e);
+    }
+  }
+
+  /**
+   * Check whether Tachyon is running in secure mode, such as SIMPLE, KERBEROS.
+   * @param authType the authentication type in configuration
+   */
+  private static void isSecurityEnabled(AuthenticationFactory.AuthType authType) {
```

Comment:
NIT: this function seems to do a validation job, rather than returning a boolean value. In this case, `isSecurityEnabled` seems less proper as it indicates returning some boolean by the usual naming convention. How about changing it to `checkSecurityEnabled`?